src/tests/ftest/datamover/posix_preserve_props.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2020-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+'''
+from data_mover_test_base import DataMoverTestBase
+from os.path import join
+from pydaos.raw import DaosApiError
+import avocado
+
+class DmvrPreserveProps(DataMoverTestBase):
+    # pylint: disable=too-many-ancestors
+    """Data Mover validation for --preserve-props option only for DFS copies.
+    Test Class Description:
+        Tests the following cases:
+            1. DAOS -> POSIX: Write container properties to an hdf5 file, and write
+               to destination posix path.
+            2. POSIX -> DAOS: Read container properties from an hdf5 file, and
+               write to destination container.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up each test case."""
+        # Start the servers and agents
+        super().setUp()
+
+        # Get the parameters
+        self.ior_flags = self.params.get(
+            "ior_flags", "/run/ior/*")
+        self.test_file = self.ior_cmd.test_file.value
+
+        # For dataset_gen and dataset_verify
+        self.obj_list = []
+
+    def run_dm_preserve_props(self, tool, cont_type, api):
+        """
+        Test Description:
+            Tests Data Mover container property preservation for DFS copies to and from POSIX.
+        Use Cases:
+            Create pool1.
+            Create a container, and write data to it.
+            Create POSIX path.
+            Copy cont1 to new posix path using --preserve-props and write container props to
+            hdf5 file.
+            Copy from posix path to cont2 and read properties from hdf5 file with --preserve-props.
+        """
+        # Set the tool to use
+        self.set_tool(tool)
+
+        # Set the api to use
+        self.set_api(api)
+
+        # Create 1 pool
+        pool1 = self.create_pool()
+        pool1.connect(2)
+
+        # set the path to read and write container properties
+        self.preserve_props_path = self.tmp + "/cont_props.h5"
+
+        # Create a source cont
+        cont1 = self.create_cont(pool1, cont_type=cont_type)
+
+        # Create source data
+        src_props = self.write_cont(cont1)
+
+        # Create a posix source path
+        posix_path = join(self.new_posix_test_path(), self.test_file)
```

Comment:
This might want to be os.path.join as well, I can't easily tell.

Reply:
@ashleypittman this is using os.path.join from the import path at top of python file:
from os.path import join